### PR TITLE
Simplify WAL to not compact

### PR DESCRIPTION
**Pull request by pauldix**
Updated the WAL to have a single partition, not compact files, and to automatically flush segment files on startup. Greatly simplifies things and should make it easier to see what is happening in the WAL.

### CODE REVIEW

File: tsdb/config.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ const (
 
 	// DefaultFlushColdInterval specifies how long after a partition has been cold
 	// for writes that a full flush and compaction are forced
-	DefaultFlushColdInterval = 5 * time.Minute
+	DefaultFlushColdInterval = 5 * time.Second
```
Comment (otoolep) [inline tsdb/config.go]: This is on purpose? I just want to confirm. It's a 60x difference.

Comment (pauldix) [inline tsdb/config.go]: yes on purpose, no reason to keep stuff cached in memory for long. If they're only writing once every 5 seconds, they could bypass the wal anyway.

File: tsdb/engine/wal/wal.go [hunk 2]
```diff
@@ -569,22 +518,19 @@ func (l *Log) Close() error {
...
 
 // close all the open Log partitions and file handles
 func (l *Log) close() error {
-	for _, p := range l.partitions {
-		if err := p.Close(); err != nil {
-			// log and skip so we can close the other partitions
-			l.logger.Println("error closing partition:", err)
-		}
+	if err := l.partition.Close(); err != nil {
+		// log and skip so we can close the other partitions
```
Comment (otoolep) [inline tsdb/engine/wal/wal.go]: Nit: this comment is no longer accurate.

File: tsdb/engine/wal/wal.go [hunk 3]
```diff
@@ -827,17 +756,28 @@ func (p *Partition) Close() error {
...
 		if p.memorySize > p.sizeThreshold {
-			return true
+			if !p.compactionRunning {
+				shouldCompact = true
+			}
+			shouldSleep = true
+
+			// sleep for an increasing amount of time based on the percentage over the memory threshold we are
+			// over := float64(p.memorySize-p.sizeThreshold) / float64(p.sizeThreshold) * float64(100)
+			sleepTime = 0 // time.Duration(2*int(over)) * time.Millisecond
```
Comment (otoolep) [inline tsdb/engine/wal/wal.go]: Where does the magic 2 come from?

**Comment by otoolep** [comment]
Some small feedback, nothing bad jumps out, but my understanding is not super-deep. +1

File: tsdb/engine/wal/wal.go (hunk 3 above)
Comment (pauldix) [inline tsdb/engine/wal/wal.go]: was totally arbitrary, but I commented it out for now.

**Comment by pauldix** [comment]
Going to merge this to get into the nightly

**Pull request merged**